HW1/P6/P6.py

```diff
@@ -0,0 +1,82 @@
+
+# coding: utf-8
+
+# In[1]:
+
+import findspark
+import os
+findspark.init('/home/chongmo/spark') # you need that before import pyspark.
+import pyspark
+from pyspark import SparkContext
+sc =SparkContext()
+
+
+# In[3]:
+
+S_RDD= sc.textFile('100.txt')
+
+
+# In[6]:
+
+#filter out and give words index
+W_RDD=S_RDD.flatMap(lambda x: x.split()).filter(lambda x: (not x.isdigit()) & (not x.isupper()) & (not x.replace('.','').isupper())).zipWithIndex().cache()
+
+
+# In[9]:
+
+# use index/3 and index%3 to keep the order of all three connected words
+# Example: The Project Gutenberg EBook of The Complete Works of William Shakespeare, by William Shakespeare
+# W1_RDD gives all three words: ((u'The', u'Project'), u'Gutenberg'), ((u'EBook', u'of'), u'The'), ...
+W1_RDD=W_RDD.map(lambda x: (x[1]/3, (x[0], x[1]%3))).groupByKey().filter(lambda x: len(x[1])==3).map(lambda x: ((list(x[1])[0][0], list(x[1])[1][0]), list(x[1])[2][0]))
+
+
+# In[13]:
+
+# W2_RDD gives all three words: ((u'Project', u'Gutenberg'), u'EBook'), ((u'of', u'The'), u'Complete'), ...
+W2_RDD=W_RDD.map(lambda x: (x[1]/3+1 if x[1]%3!=0 else x[1]/3, [x[0], x[1]%3])).groupByKey().filter(lambda x: len(x[1])==3).map(lambda x: ((list(x[1])[0][0], list(x[1])[1][0]), list(x[1])[2][0]))
+
+
+# In[16]:
+
+# W3_RDD gives all three words: ((u'Gutenberg', u'EBook'),u'of'), ((u'The', u'Complete'), u'Works'), ...
+W3_RDD=W_RDD.map(lambda x: (x[1]/3+1 if x[1]%3==2 else x[1]/3, [x[0], x[1]%3])).groupByKey().filter(lambda x: len(x[1])==3).map(lambda x: ((list(x[1])[0][0], list(x[1])[1][0]), list(x[1])[2][0]))
+
+
+# In[18]:
+
+#combine the three RDD
+Triple_W_RDD=W1_RDD.union(W2_RDD).union(W3_RDD)
+
+
+# In[19]:
+
+from collections import Counter
+Markov_RDD=Triple_W_RDD.reduceByKey(lambda x, y: x + ' '+y).mapValues(lambda x: Counter(x.split()).items())
```

Comment:
It's not clear to me what you're using `Counter` for here. Is there a way you could do this with RDD transformations rather than going into the `collections` module?